client/my-sites/patterns/components/pattern-gallery/client.tsx

```diff
@@ -184,6 +185,7 @@ export const PatternGalleryClient: PatternGalleryFC = ( props ) => {
 										type: getTracksPatternType( patternTypeFilter ),
 										view: isGridView ? 'grid' : 'list',
 										load_more_page: currentPage,
+										search_term: searchTerm,
```

Comment:
```suggestion
										search_term: searchTerm ? searchTerm : undefined,
```
This way, we won't be sending an empty `search_term`. Perhaps we can do the same for `category`.

Reply:
Makes sense! I also found an existing `searchTerm` event prop that was using `||` instead of a ternary. I've replaced that one with a ternary as well, in the interest of consistency.

re: treating `category` the same way, there are a handful of them sprinkled throughout the pattern library, all of which are would need updating. Are you happy rolling that into this PR? Or would you prefer I put up a quick, separate one?

Reply:
Ah, good eye! The `||` approach actually looks cleaner, I didn't think of that! Let's go with `search_term: searchTerm || undefined`. And we can treat `category` the same way for `calypso_pattern_library_load_more` to keep this PR in scope. We have another PR https://github.com/Automattic/wp-calypso/pull/89270 which also touches on analytics, we might be able to address it there.

Reply:
Thanks @wongasy!

I've updated this PR and will deploy once CI clears again.